### PR TITLE
Spotify API

### PR DESCRIPTION
**Pull request by emilehreich**
In this work I setup all the dependencies required to link our application to Spotify.

I use the Spotify developer tools (Spotify SDK) (caveat: still in Beta mode):
it consists of the following:

- Spotify Authentication Library: This library provides a way to obtain OAuth access tokens that can subsequently be used to play music or used in calls to the Spotify Web API.
- Spotify App Remote Library: This library contains classes for playback control and metadata access.

I setup the Spotify App Remtote Library for starters (it should be enough given the needed functionalities). For this I added a libs directory in which I put the .aar archive of the library and added it to the dependencies.

In the Model, I updated the enum structure to include  the CLIENT_ID and REDIRECT_URI which are required to connect to spotify. (This is under the assumption that DEEZER will have a similar structure, but this can be updated to be more generic in the future if needed).

The MainActivty2 is only for demo, it will just play a random playlist by calling spotify as a service.

### CODE REVIEW

File: app/src/main/java/com/github/orkest/View/MainActivity2.kt [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+/**
+ * Activity used to test the Spotify API (DEMO ONLY)
+ */
+class MainActivity2 : AppCompatActivity() {
+    private var mSpotifyAppRemote: SpotifyAppRemote? = null
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main2)
+    }
```
Comment (saraanej) [inline app/src/main/java/com/github/orkest/View/MainActivity2.kt:21]: Have you chosen to use xml or are you going back to JetPack Compose for this activity ? (curious)

File: app/src/main/java/com/github/orkest/View/MainActivity2.kt [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+import com.github.orkest.R
+import com.spotify.android.appremote.api.ConnectionParams
+import com.spotify.android.appremote.api.Connector.ConnectionListener
+import com.spotify.android.appremote.api.SpotifyAppRemote
+import com.spotify.protocol.types.PlayerState
+
+/**
+ * Activity used to test the Spotify API (DEMO ONLY)
+ */
+class MainActivity2 : AppCompatActivity() {
```
Comment (saraanej) [inline app/src/main/java/com/github/orkest/View/MainActivity2.kt:16]: LGTM

File: app/src/main/java/com/github/orkest/View/MainActivity2.kt [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        SpotifyAppRemote.connect(this, connectionParams,
+            object : ConnectionListener {
+                override fun onConnected(spotifyAppRemote: SpotifyAppRemote) {
+                    mSpotifyAppRemote = spotifyAppRemote
+                    Log.d("MainActivity", "Connected! Yay!")
+
+                    // Now you can start interacting with App Remote
+                    connected()
+                }
+
```
Comment (saraanej) [inline app/src/main/java/com/github/orkest/View/MainActivity2.kt:39]: Thank you for the comments

**Review by saraanej — APPROVED** [review]
Thank you for your code. Everything looks good to me, concise and precise.

File: app/src/main/java/com/github/orkest/View/MainActivity2.kt [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+            .subscribeToPlayerState()
+            .setEventCallback { playerState: PlayerState ->
+                val track = playerState.track
+                if (track != null) {
+                    Log.d("MainActivity", track.name + " by " + track.artist.name)
+                }
+            }
+    }
+
+    companion object {
```
Comment (evguenirousselot) [inline app/src/main/java/com/github/orkest/View/MainActivity2.kt:68]: Nice that you added a companion object and not static keywords to respect the language design

File: app/src/main/java/com/github/orkest/View/MainActivity2.kt [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+        private const val CLIENT_ID = "e7ac920406d54975bc79962dec94f4ab"
+        private const val REDIRECT_URI = "https://com.github.orkest/callback/"
+    }
```
Comment (evguenirousselot) [inline app/src/main/java/com/github/orkest/View/MainActivity2.kt:71]: Nice that you added a companion object and not static keywords to respect the language design

**Comment by evguenirousselot** [comment]
Did you manage to find a way to simulate the proper connection to Spotify's API or just like with google auth, calling the API of a third party can only be tested manually ?

**Comment by evguenirousselot** [comment]
And generally very good code ;)

File: app/src/main/java/com/github/orkest/View/MainActivity2.kt (hunk 1 above)
Comment (evguenirousselot) [reply app/src/main/java/com/github/orkest/View/MainActivity2.kt:21]: That was just for testing so he did it as quick as possible

**Pull request merged**